### PR TITLE
Handle growth.sapling for spigot config in Mangrove Propagule

### PR DESCRIPTION
**Pull request by Doc94**
Based in https://github.com/PaperMC/Paper/pull/8023#issuecomment-1160501192 this PR handle the spigot config for saplings in the mangrove.

### CODE REVIEW

File: patches/server/0918-Handle-growth.sapling-for-spigot-config-in-Mangrove-.patch [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+diff --git a/src/main/java/net/minecraft/world/level/block/MangrovePropaguleBlock.java b/src/main/java/net/minecraft/world/level/block/MangrovePropaguleBlock.java
+index 4eaf75ef7362ee781762f5e8def6bc4d986e9db7..ea1ed2e05f88ab591db45db6c1f8e51eaab4f59a 100644
+--- a/src/main/java/net/minecraft/world/level/block/MangrovePropaguleBlock.java
++++ b/src/main/java/net/minecraft/world/level/block/MangrovePropaguleBlock.java
+@@ -89,7 +89,7 @@ public class MangrovePropaguleBlock extends SaplingBlock implements SimpleWaterl
+     @Override
+     public void randomTick(BlockState state, ServerLevel world, BlockPos pos, RandomSource random) {
+         if (!isHanging(state)) {
+-            if (random.nextInt(7) == 0) {
++            if (random.nextInt(Math.max(2, (int) (((100.0F / world.spigotConfig.saplingModifier) * 7) + 0.5F))) == 0) { // Paper - handle spigot config for saplingModifier
```
Comment (Machine-Maker) [inline patches/server/0918-Handle-growth.sapling-for-spigot-config-in-Mangrove-.patch:16]: I think this is wrong. If I set the sapling modifier to 200 (200%), it becomes a `random.nextInt(4) == 0` or a 1/4 chance. If its left to default at 100 it is a `random.nextInt(7) == 0` or a 1/7 chance. 1/4 is not twice as likely as 1/7.

**Review by Machine-Maker — CHANGES_REQUESTED** [review]

File: patches/server/0918-Handle-growth.sapling-for-spigot-config-in-Mangrove-.patch (hunk 1 above)
Comment (Doc94) [reply patches/server/0918-Handle-growth.sapling-for-spigot-config-in-Mangrove-.patch:16]: hmmm i based this in all the other place where this config is set then not sure if im understanding wrong or maybe all the places where this is used is bad calculed in first place :/

Comment (Lulu13022002) [reply patches/server/0918-Handle-growth.sapling-for-spigot-config-in-Mangrove-.patch:16]: Not all the places but some has the same issue, maybe using float is better (and more readable: no abusive int cast) like in the kelp patch 0796-Fix-kelp-modifier-changing-growth-for-other-crops.patch.
For example
if(random.nextFloat() < Math.max((1/2), ((1/7) * (modifier / 100.0F)))

**Comment by Owen1212055** [comment]
Are you still interested in resolving the comment marked above? Just want to make sure incase somebody else wants to pick this up. 😃

**Comment by Doc94** [comment]
> Are you still interested in resolving the comment marked above? Just want to make sure incase somebody else wants to pick this up. 😃

you mean change the calc by the suggest by lulu? i remember not touch this because i was confuse about the correct calc (and dont wanna rebase and remake for every try)

**Pull request closed without merging**